pkg/cmd/util_containerization.go

```diff
@@ -0,0 +1,178 @@
+	args := docker.BuildBaseImageArgs()
+	cmd := exec.CommandContext(ctx, "docker", args...)
+
+	// Output executed command.
+	fmt.Printf("Executing: " + strings.Join(cmd.Args, " ") + "\n")
+
+	// Run the command.
+	if err := cmd.Run(); err != nil {
+		return errors.Errorf("base image containerization did not run successfully: %v", err)
+	}
+
+	return nil
+}
+
+func createAndBuildIntegrationImage(containerRegistry string, justBaseImage bool, image string,
+	propertyFiles []string, dependencies []string, routes []string) error {
+	// This ensures the Dockerfile for the base image will not end up in an undesired location.
+	if docker.BaseWorkingDirectory == "" {
+		return errors.New("base directory that holds the base image Dockerfile has not been set correctly")
+	}
+
+	docker.RegistryName = containerRegistry
+	if !justBaseImage {
+		registryName, err := docker.ExtractRegistryName(image)
+		if err != nil {
+			return err
+		}
+
+		docker.RegistryName = registryName
+	}
+
+	// Create the Dockerfile and build the base image.
+	err := createAndBuildBaseImage(containerRegistry)
+	if err != nil {
+		return err
+	}
+
+	if justBaseImage {
+		return nil
+	}
+
+	if docker.IntegrationWorkingDirectory == "" {
+		return errors.New("integration directory that holds the image Dockerfile has not been set correctly")
+	}
+
+	// Create integration image if integration files were provided.
+	// Copy dependencies to a dependencies folder under a local directory.
+	err = updateIntegrationDependencies(dependencies)
+	if err != nil {
+		return err
+	}
+
+	// Copy routes to a routes folder under a local directory.
+	err = updateIntegrationRoutes(routes)
+	if err != nil {
+		return err
+	}
+
+	// Get integration run command to be run inside the container. This means the command
+	// has to be created with the paths which will be valid inside the container.
+	containerCmd := GetContainerIntegrationRunCommand(propertyFiles, dependencies, routes)
+
+	// Create the integration image Docker file.
+	err = docker.CreateIntegrationImageDockerFile(containerCmd)
+	if err != nil {
+		return err
+	}
+
+	// Get the Docker command arguments for building the base image and create the command.
+	args := docker.BuildIntegrationImageArgs(image)
+	cmd := exec.CommandContext(ctx, "docker", args...)
+
+	// Output executed command.
+	fmt.Printf("Executing: " + strings.Join(cmd.Args, " ") + "\n")
+
+	// Run the command.
+	if err := cmd.Run(); err != nil {
+		return errors.Errorf("integration image containerization did not run successfully: %v", err)
+	}
+
+	return nil
+}
+
+func runIntegrationImage(image string) error {
+	// Get the docker command line argument for running an image.
+	args := docker.RunIntegrationImageArgs(image)
+	cmd := exec.CommandContext(ctx, "docker", args...)
+
+	// Output executed command.
+	fmt.Printf("Executing: " + strings.Join(cmd.Args, " ") + "\n")
+
+	// Run the command.
+	if err := cmd.Run(); err != nil {
```

Comment:
Should you set the stdout and stderr here? It works but I see no logs

Reply:
Can definitely do that. Are there some guidelines as to how std out/err should be managed?

Reply:
I've added some simple redirection of stdout and stderr.